Practical Dependency Inversion/ItemsUIComposer.swift

```diff
@@ -35,12 +35,14 @@ class MainThreadDecorator: ItemsViewControllerSpec {
     }
     
     func items(completion: @escaping (Result<[Item], Error>) -> Void) {
-        print("it is on the same queue of caller")
-        guard (DispatchQueue.getSpecific(key: Self.mainQueueKey) == Self.mainQueueValue)
-        else {
-            DispatchQueue.main.async(execute: { self.loader.items(completion: completion)} )
-            return
+        loader.items {
+            result in
+            DispatchQueue.main.async {
```

Comment:
So, This means that there is no need to identify which the current queue is, always dispatch to main queue would be good. Am I right?
Pretty sensible way, I think.👍

OTOH
So I guess that this means that you can get rid of both [this](https://github.com/rollr76518/Practical-Dependency-Inversion/pull/1/commits/8b7983ef600c218655afdeb73145abc57ba83373#diff-404da706489f4f16c0022da0e6c41898R24-R26) and [this](https://github.com/rollr76518/Practical-Dependency-Inversion/pull/1/commits/8b7983ef600c218655afdeb73145abc57ba83373#diff-404da706489f4f16c0022da0e6c41898R30-R34)?

Reply:
Sorry, can't really understand what's that means.
> So I guess that this means that you can get rid of both this and this?

And the test code for `isOnMainQueue` is just to make sure that isMainQueue is always giving the right result.

Reply:
> This means that there is no need to identify which the current queue is, always dispatch to main queue would be good. 

It's still questionable, but it seems harmless by dispatch without checking.

Reply:
> Sorry, can't really understand what's that means.
> 
> > So I guess that this means that you can get rid of both this and this?
> 
> And the test code for `isMainQueue` is just to make sure which will always give the right result.

Just wanted to pick line 24~26, line30~34 but it ain't work at all.
U don't need to specify main queue since those main queue checking is no longer needed, do u?

Reply:
Absolutely, thanks for point out. I will remove those statements.